packages/page-claims/src/index.tsx

```diff
@@ -73,183 +68,242 @@ const Signature = styled.textarea`
-    }));
-  }
+  const statementSentence = statementKind ? getStatementSentence(statementKind) : '';
 
-  private onCopy = (): void => {
-    this.setState({ didCopy: true });
-  }
+  const prefix = u8aToString(api.consts.claims.prefix.toU8a(true));
+  const payload = accountId
+    ? `${prefix}${u8aToHex(decodeAddress(accountId), -1, false)}${statementSentence}`
+    : '';
 
-  private setStep = (step: Step): () => void =>
-    (): void => {
-      this.setState({ step });
-    }
+  return (
+    <main>
+      <header />
+      {!isOldClaimProcess && <Warning />}
+      <h1>
+        <Trans>claim your <em>{TokenUnit.abbr}</em> tokens</Trans>
+      </h1>
+      <Columar>
+        <Column>
+          <Card withBottomMargin>
+            <h3>{t('1. Select your {{chain}} account', {
+              replace: {
+                chain: systemChain
+              }
+            })}</h3>
+            <InputAddress
+              defaultValue={accountId}
+              help={t('The account you want to claim to.')}
+              label={t('claim to account')}
+              onChange={setAccountId}
+              type='all'
+            />
+            {(step === Step.Account) && (
+              <Button.Group>
+                <Button
+                  icon='sign-in'
+                  isDisabled={preclaimEthereumAddress === PRECLAIMS_LOADING}
+                  label={preclaimEthereumAddress === PRECLAIMS_LOADING
+                    ? t('Loading')
+                    : t('Continue')
+                  }
+                  onClick={handleAccountStep}
+                />
+              </Button.Group>
+            )}
+          </Card>
+          {
+            // We need to know the ethereuem address only for the new process
+            // to be able to know the statement kind so that the users can sign it
+            (step >= Step.ETHAddress && !isPreclaimed && !isOldClaimProcess) && (
+              <Card withBottomMargin>
+                <h3>{t('2. Enter the ETH address from the sale.')}</h3>
+                <Input
+                  autoFocus
+                  className='full'
+                  help={t('The the Ethereum address you used during the pre-sale (starting by "0x")')}
+                  label={t('Pre-sale ethereum address')}
+                  onChange={onChangeEthereumAddress}
+                  value={ethereumAddress || ''}
+                />
+                {(step === Step.ETHAddress) && (
+                  <Button.Group>
+                    <Button
+                      icon='sign-in'
+                      isDisabled={!ethereumAddress}
+                      label={t('Continue')}
+                      onClick={goToStepSign}
+                    />
+                  </Button.Group>
+                )}
+              </Card>
+            )}
+          {(step >= Step.Sign && !isPreclaimed) && (
+            <Card>
+              {(isOldClaimProcess || !!statementKind)
+                ? <>
```

Comment:
Because we don't use `()` around a multi-line component structure, the intent is completely lost. Improve for readability.